asserts/pool.go

```diff
@@ -478,36 +645,65 @@ func (p *Pool) Add(a Assertion, grouping Grouping) (ok bool, err error) {
 }
 
 func (p *Pool) addToGrouping(a Assertion, grouping Grouping, deserializeGrouping func(string) (*internal.Grouping, error)) (ok bool, err error) {
-	uniq := a.Ref().Unique()
-	var u *unresolvedRec
-	var extrag *internal.Grouping
-	var unresolved map[string]*unresolvedRec
-	if u = p.unresolved[uniq]; u != nil {
-		unresolved = p.unresolved
-	} else if u = p.prerequisites[uniq]; u != nil {
-		unresolved = p.prerequisites
-	} else {
-		ok, err := p.isPredefined(a.Ref())
-		if err != nil {
-			return false, err
+	var uniq string
+	ref := a.Ref()
+	// deal with sequence key
+	if ref.Type.SequenceForming() {
```

Comment:
I would prefer the if to be negated so that we take care of non-sequence forming first